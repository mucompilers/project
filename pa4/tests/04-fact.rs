fn fact(n : i32) -> i32 {
      return if (n < 1) { 1 } else { n * fact(n - 1) };
}

fn main() {
      println!("{}", fact(1));
      println!("{}", fact(5));
      println!("{}", fact(10));
}
