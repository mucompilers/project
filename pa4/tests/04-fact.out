1
120
3628800
